### ISSUE TITLE
False positives when the underscore hack is used with indentation

### ISSUE THREAD

**Issue by yisibl** [stylelint/stylelint#1787]
I found out that there was only hack for *.

https://github.com/stylelint/stylelint/blob/master/src/rules/indentation/index.js#L63

```css
.foo {
  _margin: 10px; /* Expected indentation of 2 spaces   indentation */
}
```

**Comment by evilebottnawi** [comment]
@yisibl we use https://github.com/ben-eb/stylehacks inside rule, please create issue in this repo, not here

**Comment by jeddy3** [comment]
I can reproduce this. 

@evilebottnawi This is a bug with the `indentation` rule and I don't think it's related to `stylehacks`. @yisibl Would you like to contribute a fix (and corresponding test)?. It looks to be a be a simple one.

**Comment by yisibl** [comment]
@jeddy3 No problem! 
This really should not be handed over to stylehacks.

I can fix the underscore hack problem, but should we consider more full CSS hackers?

* https://swordair.com/tools/css-hack-table/
* http://browserhacks.com/

**Comment by evilebottnawi** [comment]
@jeddy3 oh sorry

**Comment by evilebottnawi** [comment]
@jeddy3 what do you if we drop `stylehacks` and rewrite rule with https://swordair.com/tools/css-hack-table/, also rule can support `SCSS`, `Less` and etc. I can do this.

**Comment by davidtheclark** [comment]
@evilebottnawi I'm really not interested in putting work into CSS hacks.

**Comment by evilebottnawi** [comment]
@davidtheclark what about add `hasHack` to `stylehack` and rewrite rule on `walkDecl` and `walkAtRules`?

**Comment by davidtheclark** [comment]
@evilebottnawi Yeah, if you want to work on `stylehack`, then take advantage of that in `stylelint`, that's great. I just feel like we shouldn't add significant code to `stylelint` dealing with hacks.

**Comment by evilebottnawi** [comment]
@davidtheclark `stylehack` have little hacks and not support non-standard syntax, i can will work on this.
- `stylehack` ignore non standard syntax (it will be not simple)
- `stylelint` rewrite rule on `hasHack` and `walkDecl`, `walkAtRule` and etc (it will be simple)

**Comment by davidtheclark** [comment]
> stylelint rewrite rule on hasHack and walkDecl, walkAtRule and etc (it will be simple)

I don't really understand what you mean by this, but ok, will look at PR.
